contracts/Failure.sol

```diff
@@ -5,4 +5,12 @@ contract Failure {
     function fail() public payable {
         revert("big fail!");
     }
+
+    function failWithBurn(uint256 _amount) public payable {
```

Comment:
If the amount is smaller than the provided gas it doesn't fail. Maybe just rename it to `burnGas` and add a docstring it burns at least that much gas? (but not exactly)

Reply:
Renamed to `burn`.